Sanghyun/controllers/userController.js

```diff
@@ -0,0 +1,35 @@
+const userService = require("../services/userService");
+
+const createUser = async (req, res) => {
+  try {
+    const { userName, email, profileImage, userPassword } = req.body;
+
+    const pwValidation = new RegExp(
+      "^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})"
+    );
+    if (!pwValidation.test(userPassword)) {
+      const err = new Error("password is not valid");
+      err.statusCode = 400;
+      throw err;
+    }
+
+    if (!userName || !email || !profileImage || !userPassword) {
+      return res.status(400).json({ message: "Missing required fields" });
+    }
```

Comment:
```suggestion
    if (!userName || !email || !profileImage || !userPassword) {
      return res.status(400).json({ message: "Missing required fields" });
    }

    const pwValidation = new RegExp(
      "^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})"
    );
    if (!pwValidation.test(userPassword)) {
      const err = new Error("password is not valid");
      err.statusCode = 400;
      throw err;
    }
```
현재와 같은 코드 순서라면 userPassword가 들어오지 않았을 때, key error임에도 불구하고 정규식을 먼저 검증하기 때문에 클라이언트에서 정확한 문제 파악을 캐치하기 힘들 수 있으므로 위의 예시와 같은 순서로 작성되어야 할 것 같습니다.